src/util/schnorr.rs

```diff
@@ -45,6 +45,13 @@ impl fmt::Display for TweakedPublicKey {
     }
 }
 
+impl str::FromStr for TweakedPublicKey {
+    type Err = secp256k1::Error;
+    fn from_str(s: &str) -> Result<TweakedPublicKey, secp256k1::Error> {
+        Ok(TweakedPublicKey(XOnlyPublicKey::from_str(s)?))
```

Comment:
In 8b4c5b9d5d0c11b13a2f0cfbe26d20cd7cf24a86
Is it a good idea to have this? I would rather have users parse `UnTweakedPubKey` and call assume_dangerous.

Reply:
`FromStr` is required by serde impl. Also, this brings us back to the question raised in #737.

Reply:
I suggest don't impl (De)Serialize until we are not confident what the representation should be. We can always add impl later without breaking anything.

Reply:
If I am not wrong `TweakedPublicKey` should be used in PSBT, and removing serde there will be a regression

Reply:
I think it's okay to parse as `TweakedPublicKey`. I agree that there's a safety loophole here because we can't actually distinguish tweaked keys from untweaked keys, but I think it would be too unergonomic (and as @dr-orlovsky says, possibly incompatible with the PSBT API) to make the user do the `assume_dangerous` dance every time they parse a key.

Especially because untweeaked keys will not exactly be common in the wild .. so we are asking the user to think about an undetectable rare situation that they may not even understand.

Reply:
I don't see why we need TweakedPubkeys in PSBT APIs? I think partial sigs should be `XOnlyPublicKey` -> `SchnorrSig`.

Reply:
In the most common case, those XOnlyKeys would be tweaked, but I don't see any benefit for adding that type in psbt API. In my view, the _only_ purpose of `TweakedPublicKey` is to guard users using the address API. 

The keys in other places in the codebase might also be tweaked public keys, but we don't gain anything by having those being tweaked public keys (for example psbt API should only operate on XOnlyKeys).  IMO, we should just remove this commit.

Reply:
We will still need this for `rawtr` processing

Reply:
We can parse `XOnlyPubKey`, and use the `assume_dangerous_tweaked` API? I think that is preciously how `rawtr` should work (if it is added in spec). 
I feel strongly about this because we did put so much into Tweaking API, and added type safety with `TweakedPublicKey`. It would be a shame if we can break it this easily.